lfs/pointer_smudge.go

```diff
@@ -57,7 +85,22 @@ func PointerSmudge(writer io.Writer, ptr *Pointer, workingfile string, download
 
 	if statErr != nil || stat == nil {
 		if download {
-			err = downloadFile(writer, ptr, workingfile, mediafile, manifest, cb)
+			// TODO: Check if the `ptr.Oid` is already being downloaded.
+			// If no download is running then start the download right away!
+			// If a download of the `ptr.Oid` is running then check if the
+			// `workfile` is the same. If this is the case then no further
+			// action is required. If the `workfile` is not the same then
+			// ensure that the file is written on `WaitForDownloads`.
+			//
+			// Important: GitLFS should not write the files directly to the
+			// worktree when they finish. It should write the files only to the
+			// local media path. Only at the end, when Git signals GitLFS with
+			// an EOF its exit then GitLFS should write the files (in the
+			// `WaitForDownloads` step).
+			var dd DeferredDownload
+			dd.Ptr = ptr
+			dd.WorkingFile = workingfile
+			deferredDownloads = append(deferredDownloads, dd)
```

Comment:
Here we write nothing to `writer` which means the `smudge` result is an empty file. We might want to write the GitLFS pointer there to `writer` or a message like `file is being downloaded`. It doesn't matter what we write ... the file is overwritten at the end anyways. This intermedia step is only relevant in case GitLFS dies...